docs/reference/alpha-vector-database.md

```diff
@@ -0,0 +1,96 @@
+# [Alpha] Vector Database
+**Warning**: This is an _experimental_ feature. To our knowledge, this is stable, but there are still rough edges in the experience. Contributions are welcome!
+
+## Overview
+Vector database allows user to store and retrieve embeddings. Feast provides general APIs to store and retrieve embeddings.
+
+## Integration
+Below are supported vector databases and implemented features:
+
+| Vector Database | Retrieval | Indexing |
+|-----------------|-----------|----------|
+| Pgvector        | [x]       | [ ]      |
+| Elasticsearch   | [ ]       | [ ]      |
+| Milvus          | [ ]       | [ ]      |
+| Faiss           | [ ]       | [ ]      |
+
+
+## Example
+
+See [https://github.com/feast-dev/feast-workshop/blob/rag/module_4_rag](https://github.com/feast-dev/feast-workshop/blob/rag/module_4_rag) for an example on how to use vector database.
+
+### **Prepare offline embedding dataset**
+Run the following commands to prepare the embedding dataset:
+```shell
+python pull_states.py
+python batch_score_documents.py
+```
+The output will be stored in `data/city_wikipedia_summaries.csv.`
+
+### **Initialize Feast feature store and materialize the data to the online store**
+Use the feature_tore.yaml file to initialize the feature store. This will use the data as offline store, and Pgvector as online store.
+
+```yaml
+project: feast_demo_local
+provider: local
+registry:
+  registry_type: sql
+  path: postgresql://@localhost:5432/feast
+online_store:
+  type: postgres
+  pgvector_enabled: true
+  vector_len: 384
+  host: 127.0.0.1
+  port: 5432
+  database: feast
+  user: ""
+  password: ""
+
+
+offline_store:
+  type: file
+entity_key_serialization_version: 2
+```
+Run the following command to apply the feature store configuration:
+
+```shell
+feast apply
+```
+
```

Comment:
```suggestion

Note that when you run `feast apply` you are going to apply the following Feature View that you will use for retrieval later:
```python
city_embeddings_feature_view = FeatureView(
    name="city_embeddings",
    entities=[item],
    schema=[
        Field(name="Embeddings", dtype=Array(Float32)),
    ],
    source=source,
    ttl=timedelta(hours=2),
)
```

```

Reply:
make sense. Updated the doc